### PR TITLE
Makes the router responsible for setting the router_http_host config …

### PR DESCRIPTION
**Pull request by navonod**
…variable

I'm not sure about the generated go.mod & go.sum stuff - please look-see

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -80,6 +80,14 @@ func NewServer(config *Config) (*Server, error) {
 		return s.translator.getProxy(s.errorLog, req.URL.Host)
 	}
 
+	// Set both the host and port as system config variables
+	hostname,err := os.Hostname()
```
Comment (LeonLeibbrandt) [inline server/server.go:84]: Just check if this will also work in docker.

**Review by LeonLeibbrandt — APPROVED** [review]
LGTM

File: server/server.go (hunk 1 above)
Comment (navonod) [reply server/server.go:84]: I tested a router to config service call already, using the `whoami` result as the host name, which worked. Will also test the code change.

Comment (navonod) [reply server/server.go:84]: Confirmed: working in docker too

**Comment by bmharper** [comment]
I know this comment comes in too late, BUT - what it would have been nice to see an explanation of what's going on here, covering these points:
1. What is `router_http_host` used for?
2. Why the change?

**Comment by navonod** [comment]
@bmharper - will answer retrospectively:

1. Containerised asset services rely on `router_http_host` in conjunction with `router_http_port` to know where to find the hostname of the router, and
2. The previous solution for setting the host value was to use docker-compose to set it manually, but this forced a separation from where the router port gets set, and seemed cumbersome if the router could automate the process without the need for conf.

**Pull request merged**